classes/local/rule/rule_base.php

```diff
@@ -42,6 +50,46 @@ abstract class rule_base implements rule_interface {
      * @param stdClass $config rule instance configuration
      */
     public function __construct(stdClass $config) {
+        $this->config = $config;
+        $this->result = new rule_check_result();
+    }
+
+    /**
+     * Return a result indicating this check will allow user registration.
+     *
+     * @return rule_check_result
+     */
+    public function allow(): rule_check_result {
+        $this->result->set_allowed(true);
+
+        return $this->result;
+    }
+
+    /**
+     * Return a result indicating this check will deny user registration.
+     *
+     * @param int $score
+     * @param string $feedbackmessage
+     * @param array $validationmessages
+     * @throws coding_exception
+     *
+     * @return rule_check_result
+     */
+    public function deny(
+        int $score,
+        string $feedbackmessage = '',
+        array $validationmessages = []
+    ): rule_check_result {
```

Comment:
~~In your next commit implementing scoring system in disposable mail rule, you simply take the score from the `config` property, so if we don't have to override the~~

I've now seen it's used for the fallback score in case of failure. So all good here! Or we could think about using a separate method for fallback denial… :thinking: We could still refactor that later, if we wish.

Reply:
As you mention at the moment I've just implemented a single `deny()` method and pass that either `points` or `fallbackpoints`. But we could in future find a way to implement a `denyfallback()` in `rule_base` so scores don't need to be passed as params in this way.